dm/worker/config.go

```diff
@@ -262,13 +271,19 @@ func (c *Config) adjustFlavor() error {
 	if err != nil {
 		return err
 	}
-	fromDB, err := applyNewBaseDB(clone.From)
+	from := clone.From
+	from.RawDBCfg = config.DefaultRawDBConfig().SetReadTimeout(flavorReadTimeout)
+	fromDB, err := applyNewBaseDB(from)
 	if err != nil {
 		return terror.WithScope(err, terror.ScopeUpstream)
 	}
 	defer fromDB.Close()
 
-	c.Flavor, err = utils.GetFlavor(fromDB.DB)
+	ctx, _ := context.WithTimeout(context.Background(), flavorGetTimeout)
+	c.Flavor, err = utils.GetFlavor(ctx, fromDB.DB)
+	if ctx.Err() != nil {
+		err = terror.Annotate(err, "time cost to get flavor info exceeds flavorGetTimeout")
+	}
```

Comment:
Can these lines be removed? and only use the L287?

If you want to use this message, it's better to replace `flavorGetTimeout` with the accurate timeout value.

Reply:
These lines are added because when ctx is canceled because of timeout the returned error will just be like `context deadline exceeded` which lacks much information.

Reply:
fine. so how about updating `flavorGetTimeout` to a meaningful value?

Reply:
Sure, I will revise that.